### ISSUE TITLE
Split content fail column to pin_fail and storage_fail columns

### ISSUE THREAD

**Issue by en0ma** [application-research/estuary#335]
Add `pin_failed` and `storage_failed` columns to the `contents` table, to mark content IPFS pinning failure use and content Filecoin deal making failure, so we stop retrying(pinning or making deals) such contents. This will help reduce error logs spams and stressing miners.

E.g;
- If we could not fetch blocks for pinning - `pin_failed` (UI should not show it, or show it and indicate an error). 
- If Graphsync or boost irrecoverable error (commp, missing block etc) - `storage_failed`, (UI should not show it, or show it and indicate an error)

Also, given network fluctuations (bitswap), pinning can fail intermittently, we should have a `pinning_retry` count column before marking `pin_failed`.

**Comment by snissn** [comment]
makes sense! I have a little back log then we can connect on this, maybe you can onboard me to completing the ticket next week
